src/core/server/server.ts

```diff
@@ -78,6 +78,10 @@ export class Server {
     // Discover any plugins before continuing. This allows other systems to utilize the plugin dependency graph.
     const pluginDependencies = await this.plugins.discover();
     const legacyPlugins = await this.legacy.discoverPlugins();
+
+    // Immediately terminate in case of invalid configuration
+    await ensureValidConfiguration(this.configService, legacyPlugins);
```

Comment:
I used a pure function for that. Basically reasons are:

- I could have performed that directly during `this.legacy.discoverPlugins()` as we already got everything we need here, but I wanted something more explicit.
- I could have created a `this.legacy.ensureValidConfiguration()` but I didn't like to handle that kind of responsibility inside the `legacy` module
- I could have added a method in the `configService`, but I was polluting the config module with imports from legacy and I didn't like that.

Tell me if another solution is preferable.